backend/src/filter/validateMatchAllCondition.ts

```diff
@@ -0,0 +1,23 @@
+import { ArgumentError } from '../shared/errors';
+import { FilterCondition, MatchAllCondition } from '../shared/queries';
+import { formatErrorMessage } from './formatErrorMessage';
+import { checkValidMatchAllAnyCondition } from './checkValidMatchAllAnyCondition';
+
+export function validateMatchAllCondition(
+  condition: FilterCondition
+): MatchAllCondition {
+  const result = <MatchAllCondition>condition;
```

Comment:
Why is there an instant type change? So why is ne argument type not of type `MatchAllCondition`

Reply:
Thats because the function does in fact a protected cast. The check is in the next line.